### PR TITLE
EncounterGUI is fully implemented. 

### PR DESCRIPTION
**Pull request by KennyCSUEB**
The wild pokemon encounter can now be fully visualized by featuring menu control: fight moves, bag, switch pokemon, and run. Some housecleaning is likely.

### CODE REVIEW

File: pokemon/src/main/java/gamarket/EncounterGUI.java [hunk 1]
```diff
@@ -1,125 +1,1075 @@
...
+     * Updates the Wild's pokemon health, name, and level. 
+     */
+    public String updateHP2()
+    {
+        // update the wild's pokemon status bars: name, level, health. 
+        return wildPokemon.getIdentStats().getName() +
+                "       Lv." + Integer.toString(wildPokemon.getIdentStats().getLevel()) 
+                + "\n    " + Integer.toString(wildPokemon.getDefensiveStats().getHPCurrent()) + " / "
+                + Integer.toString(wildPokemon.getDefensiveStats().getHP()) + " HP";
+    }
```
Comment (ChiralAlchemist) [inline pokemon/src/main/java/gamarket/EncounterGUI.java]: these two functions are basically the same except the pokemon used. make updateHP(Pokemon poke) to make code drier

File: pokemon/src/main/java/gamarket/EncounterGUI.java [hunk 2]
```diff
@@ -1,125 +1,1075 @@
...
+        gp.setStyle("-fx-alignment: center-right;");
+        
+        // Style them health bars and add them to the main GridPane GUI 'gp'.
+        StackPane playerPokeHealthBar = new StackPane();        // The player pokemon. 
+        playerPokeHealthBar.getChildren().add(hp2);
+        gp.add(playerPokeHealthBar, 1, 0);
+        playerPokeHealthBar.setPadding(new Insets(0, 0, 532, 300));
+        hp2.setStyle("-fx-background-radius: 300px;" + "-fx-pref-width: 300px; " + "-fx-border: none; "
+            + "-fx-background-color: white;" + "-fx-border-radius: 300px;" + "-fx-border-width: 7px;"
+            + "-fx-font-size: 24px;" + "-fx-text-align: center;" + "-fx-text-base-color: rgb(0,0,0);");
```
Comment (ChiralAlchemist) [inline pokemon/src/main/java/gamarket/EncounterGUI.java]: the styles for hp2 and hp1 are the same value should be stored as a variable hpStyle

File: pokemon/src/main/java/gamarket/EncounterGUI.java [hunk 3]
```diff
@@ -1,125 +1,1075 @@
...
+        BackgroundSize bSize = new BackgroundSize(BackgroundSize.AUTO, 
+                                                    BackgroundSize.AUTO, false, false, true, true);
+        gp.setBackground(new Background(new BackgroundImage(image3, BackgroundRepeat.NO_REPEAT,
+                           BackgroundRepeat.NO_REPEAT, BackgroundPosition.CENTER, bSize)));
+ 
+        // Image for the fight button. 
+        filePath = "./pokemon/imgs/fight_button_image.png";
+        File test = new File(filePath);
+        String x = test.toURI().toString();
+        ImageView a = new ImageView(x);
```
Comment (ChiralAlchemist) [inline pokemon/src/main/java/gamarket/EncounterGUI.java]: this code block gets used a number of times. make into a function that takes a string filepath and returns an imageview

File: pokemon/src/main/java/gamarket/EncounterGUI.java [hunk 4]
```diff
@@ -1,125 +1,1075 @@
...
+ 
+        // Sync the Player's pokemon moves into the buttons.
+        move1 = new Button(); 
+        move1.setOnAction(e -> {this.move1();});
+        move2 = new Button(); 
+        move2.setOnAction(e -> {this.move2();});
+        move3 = new Button(); 
+        move3.setOnAction(e -> {this.move3();});
+        move4 = new Button(); 
+        move4.setOnAction(e -> {this.move4();});
```
Comment (ChiralAlchemist) [inline pokemon/src/main/java/gamarket/EncounterGUI.java]: the move functions should be one function that takes an int moveNumber(int buttonNum)

File: pokemon/src/main/java/gamarket/EncounterGUI.java [hunk 5]
```diff
@@ -1,125 +1,1075 @@
...
+                + "-fx-text-base-color: rgb(0,0,0);");
+        }});
+        move4.addEventHandler(MouseEvent.MOUSE_EXITED, new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent e) {
+                move4.setStyle("-fx-background-radius: 400px;" + "-fx-pref-width: 400px; " + "-fx-border: none; "
+                + "-fx-border-radius: 400px;" + "-fx-border-width: 7px;"  + "-fx-border-color: slategray;"
+                + EncounterGUI.setColor(3) + "-fx-font-size: 32px;" + "-fx-text-align: center;"
+                + "-fx-text-base-color: rgb(0,0,0);");
+        }});
```
Comment (ChiralAlchemist) [inline pokemon/src/main/java/gamarket/EncounterGUI.java]: alot of these mouse exit and mouse enter styles look to be the same. If so use the same variable.

File: pokemon/src/main/java/gamarket/EncounterGUI.java [hunk 6]
```diff
@@ -1,125 +1,1075 @@
...
+            playerPokemon = aEncounter.getPlayerTeam().switchPokemonInTeam(1, 6);
+            playerPokemon = aEncounter.getPlayerActivePokemon();
+            this.styleMoveButtons();
+            // update the pokemon status bars: name, level, health.
+            hp1.setText(this.updateHP1());
+            // change visibility of menus. 
+            battleMenu.setVisible(true);
+            pokemonMenu.setVisible(false);
+        }
+    }
```
Comment (ChiralAlchemist) [inline pokemon/src/main/java/gamarket/EncounterGUI.java]: pass in the index when call back is called not event will make codes simpler

**Review by ChiralAlchemist — CHANGES_REQUESTED** [review]
Functionally seems good. But there are a lot of spots where code is repeated where it doesn't need to be. This will make things harder to change and maintain. Dry up code and it should be good to go.

**Review by ChiralAlchemist — APPROVED** [review]
Seems good a lot less redundancy.

**Review by davidlserrano — APPROVED** [review]

**Pull request merged**